include/mbedtls/pk.h

```diff
@@ -222,7 +222,7 @@ typedef struct mbedtls_pk_context {
     void *MBEDTLS_PRIVATE(pk_ctx);                        /**< Underlying public key context  */
     /* The following field is used to store the ID of a private key in the
      * following cases:
-     * - opaque key when MBEDTLS_USE_PSA_CRYPTO is defined
+     * - opaque key when MBEDTLS_PSA_CRYPTO_CLIENT is defined
      * - normal key when MBEDTLS_PK_USE_PSA_EC_DATA is defined. In this case:
      *    - the pk_ctx above is not not used to store the private key anymore.
```

Comment:
I was going to say we'd need an extra field to distinguish between the two, but actually no: the information is already there in `pk_info`. That is, if I'm not mistaken: if type is opaque then use `priv_id`, else if type is RSA (including alt) use `pk_ctx` else depends on `USE_PSA_EC_DATA`.

As you said, it's pre-existing, and it smells like a potential can of worms to me, so I'm inclined to not even try it here, and only consider it for a follow-up. (Perhaps not in 3.6.x though due to ABI concerns?)

Reply:
Right. Definitely not try it here due to the risk, and can't do it in 3.6.x because it would change the ABI. I was asking because I was wondering if I was missing something. I find the arrangement of the “new” pk context layouts a little weird.